File: python/7_kyu/mumbling.py
# Mumbling
# https://www.codewars.com/kata/5667e8f4e3f572a8f2000039/train/python

def accum(s):
    lowercase = 0
    ans = []
    
    for i in range(0, len(s)):
            ans.append(s[i].upper() + (s[i].lower() * lowercase))
            lowercase += 1
            if i != len(s) - 1:
                ans.append("-")
    
    return ''.join(ans)

print(accum("ZpglnRxqenU"))

File: javascript/7_kyu/mumbling.js
// Mumbling
// https://www.codewars.com/kata/5667e8f4e3f572a8f2000039/train/javascript

function accum(s) {
    var lowercase = 0;
  var ans = [];
  
  for (var i = 0; i < s.length; i++) {
      var x = s[i].toUpperCase();
      var y = s[i].toLowerCase().repeat(lowercase);
      lowercase++;

      ans.push(x.concat(y));
  }
  
  return ans.join('-');
}

console.log(accum("ZpglnRxqenU"))